### PR TITLE
[libc] Update libc_errno to work correctly in both overlay and full build modes.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

**Comment by lntue** [comment]
This was part of https://github.com/llvm/llvm-project/pull/79319

File: libc/src/errno/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,20 @@
+# If we are in full build mode, we will provide the errno definition ourselves,
+# and if we are in overlay mode, we will just re-use the system's errno.
+# We are passing LIBC_FULL_BUILD flag in full build mode so that the
+# implementation of libc_errno will know if we are in full build mode or not.
+set(full_build_flag "")
```
Comment (jhuber6) [inline libc/src/errno/CMakeLists.txt:5]: Wondering if this shouldn't be top-level like the stuff in `__support/macros` and set similar to other config options.

Comment (lntue) [reply libc/src/errno/CMakeLists.txt:5]: I think @michaelrj-google also has some usage for full build mode detection inside the implementations, so I will probably move this to `_get_common_compile_options` later.

Comment (jhuber6) [reply libc/src/errno/CMakeLists.txt:5]: Alright, put a TODO here then.

**Review by jhuber6 — APPROVED** [review]
This has the effect of making the GPU `errno` externally visible. I remember some discussions with Siva about that. I don't think it will be a huge issue, at least not righ right now. Though most of the GPU implementations of `stdio` stuff intentionally ignored errno so it won't be consistent in any case. FWIW I think a global config to ignore errno might be valuable, even if it's against the standard.

File: libc/src/errno/CMakeLists.txt (hunk 1 above)
Comment (lntue) [reply libc/src/errno/CMakeLists.txt:7]: Done.

**Pull request merged**